lib/rules/valid-api.js

```diff
@@ -68,7 +68,7 @@ function validatePropertyName(property, context) {
         });
     }
 
-    if (name.startsWith('data')) {
+    if (name.match(/data.+/)) {
```

Comment:
```suggestion
    if (name.startsWith('data') && name !== 'data') {
```
`/data.+/` will be true for `undatable`.